Microsoft.Toolkit.Uwp.UI.Controls/Menu/MenuItem.cs

```diff
@@ -338,5 +411,104 @@ protected override void OnGotFocus(RoutedEventArgs e)
             _parentMenu.SelectedMenuItem = this;
             base.OnGotFocus(e);
         }
+
+        internal void Underline()
+        {
+            if (_originalHeader == null)
+            {
+                return;
+            }
+
+            var underlineCharacterIndex = _originalHeader.IndexOf(UnderlineCharacter);
+
+            var underlinedCharacter = _originalHeader[underlineCharacterIndex + 1];
```

Comment:
It will be removed in method HeaderPropertyChanged at line 496 :) so there is no need to check here again.